// devextreme-vue/drawer.js
/*!
 * devextreme-vue
 * Version: 20.1.6
 * Build date: Fri Jul 17 2020
 *
 * Copyright (c) 2012 - 2020 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var VueType = require("vue");
var Vue = VueType.default || VueType;
var drawer_1 = require("devextreme/ui/drawer");
var component_1 = require("./core/component");
var DxDrawer = Vue.extend({
    extends: component_1.DxComponent,
    props: {
        activeStateEnabled: Boolean,
        animationDuration: Number,
        animationEnabled: Boolean,
        closeOnOutsideClick: [Boolean, Function],
        disabled: Boolean,
        elementAttr: Object,
        height: [Function, Number, String],
        hint: String,
        hoverStateEnabled: Boolean,
        maxSize: Number,
        minSize: Number,
        onDisposing: Function,
        onInitialized: Function,
        onOptionChanged: Function,
        opened: Boolean,
        openedStateMode: String,
        position: String,
        revealMode: String,
        rtlEnabled: Boolean,
        shading: Boolean,
        template: {},
        visible: Boolean,
        width: [Function, Number, String]
    },
    computed: {
        instance: function () {
            return this.$_instance;
        }
    },
    beforeCreate: function () {
        this.$_WidgetClass = drawer_1.default;
    }
});
exports.DxDrawer = DxDrawer;
exports.default = DxDrawer;
