// devextreme-vue/index.js
/*!
 * devextreme-vue
 * Version: 20.1.6
 * Build date: Fri Jul 17 2020
 *
 * Copyright (c) 2012 - 2020 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var accordion_1 = require("./accordion");
exports.DxAccordion = accordion_1.DxAccordion;
var action_sheet_1 = require("./action-sheet");
exports.DxActionSheet = action_sheet_1.DxActionSheet;
var autocomplete_1 = require("./autocomplete");
exports.DxAutocomplete = autocomplete_1.DxAutocomplete;
var bar_gauge_1 = require("./bar-gauge");
exports.DxBarGauge = bar_gauge_1.DxBarGauge;
var box_1 = require("./box");
exports.DxBox = box_1.DxBox;
var bullet_1 = require("./bullet");
exports.DxBullet = bullet_1.DxBullet;
var button_1 = require("./button");
exports.DxButton = button_1.DxButton;
var button_group_1 = require("./button-group");
exports.DxButtonGroup = button_group_1.DxButtonGroup;
var calendar_1 = require("./calendar");
exports.DxCalendar = calendar_1.DxCalendar;
var chart_1 = require("./chart");
exports.DxChart = chart_1.DxChart;
var check_box_1 = require("./check-box");
exports.DxCheckBox = check_box_1.DxCheckBox;
var circular_gauge_1 = require("./circular-gauge");
exports.DxCircularGauge = circular_gauge_1.DxCircularGauge;
var color_box_1 = require("./color-box");
exports.DxColorBox = color_box_1.DxColorBox;
var context_menu_1 = require("./context-menu");
exports.DxContextMenu = context_menu_1.DxContextMenu;
var data_grid_1 = require("./data-grid");
exports.DxDataGrid = data_grid_1.DxDataGrid;
var date_box_1 = require("./date-box");
exports.DxDateBox = date_box_1.DxDateBox;
var defer_rendering_1 = require("./defer-rendering");
exports.DxDeferRendering = defer_rendering_1.DxDeferRendering;
var diagram_1 = require("./diagram");
exports.DxDiagram = diagram_1.DxDiagram;
var draggable_1 = require("./draggable");
exports.DxDraggable = draggable_1.DxDraggable;
var drawer_1 = require("./drawer");
exports.DxDrawer = drawer_1.DxDrawer;
var drop_down_box_1 = require("./drop-down-box");
exports.DxDropDownBox = drop_down_box_1.DxDropDownBox;
var drop_down_button_1 = require("./drop-down-button");
exports.DxDropDownButton = drop_down_button_1.DxDropDownButton;
var file_manager_1 = require("./file-manager");
exports.DxFileManager = file_manager_1.DxFileManager;
var file_uploader_1 = require("./file-uploader");
exports.DxFileUploader = file_uploader_1.DxFileUploader;
var filter_builder_1 = require("./filter-builder");
exports.DxFilterBuilder = filter_builder_1.DxFilterBuilder;
var form_1 = require("./form");
exports.DxForm = form_1.DxForm;
var funnel_1 = require("./funnel");
exports.DxFunnel = funnel_1.DxFunnel;
var gallery_1 = require("./gallery");
exports.DxGallery = gallery_1.DxGallery;
var gantt_1 = require("./gantt");
exports.DxGantt = gantt_1.DxGantt;
var html_editor_1 = require("./html-editor");
exports.DxHtmlEditor = html_editor_1.DxHtmlEditor;
var linear_gauge_1 = require("./linear-gauge");
exports.DxLinearGauge = linear_gauge_1.DxLinearGauge;
var list_1 = require("./list");
exports.DxList = list_1.DxList;
var load_indicator_1 = require("./load-indicator");
exports.DxLoadIndicator = load_indicator_1.DxLoadIndicator;
var load_panel_1 = require("./load-panel");
exports.DxLoadPanel = load_panel_1.DxLoadPanel;
var lookup_1 = require("./lookup");
exports.DxLookup = lookup_1.DxLookup;
var map_1 = require("./map");
exports.DxMap = map_1.DxMap;
var menu_1 = require("./menu");
exports.DxMenu = menu_1.DxMenu;
var multi_view_1 = require("./multi-view");
exports.DxMultiView = multi_view_1.DxMultiView;
var nav_bar_1 = require("./nav-bar");
exports.DxNavBar = nav_bar_1.DxNavBar;
var number_box_1 = require("./number-box");
exports.DxNumberBox = number_box_1.DxNumberBox;
var pie_chart_1 = require("./pie-chart");
exports.DxPieChart = pie_chart_1.DxPieChart;
var pivot_grid_1 = require("./pivot-grid");
exports.DxPivotGrid = pivot_grid_1.DxPivotGrid;
var pivot_grid_field_chooser_1 = require("./pivot-grid-field-chooser");
exports.DxPivotGridFieldChooser = pivot_grid_field_chooser_1.DxPivotGridFieldChooser;
var polar_chart_1 = require("./polar-chart");
exports.DxPolarChart = polar_chart_1.DxPolarChart;
var popover_1 = require("./popover");
exports.DxPopover = popover_1.DxPopover;
var popup_1 = require("./popup");
exports.DxPopup = popup_1.DxPopup;
var progress_bar_1 = require("./progress-bar");
exports.DxProgressBar = progress_bar_1.DxProgressBar;
var radio_group_1 = require("./radio-group");
exports.DxRadioGroup = radio_group_1.DxRadioGroup;
var range_selector_1 = require("./range-selector");
exports.DxRangeSelector = range_selector_1.DxRangeSelector;
var range_slider_1 = require("./range-slider");
exports.DxRangeSlider = range_slider_1.DxRangeSlider;
var recurrence_editor_1 = require("./recurrence-editor");
exports.DxRecurrenceEditor = recurrence_editor_1.DxRecurrenceEditor;
var resizable_1 = require("./resizable");
exports.DxResizable = resizable_1.DxResizable;
var responsive_box_1 = require("./responsive-box");
exports.DxResponsiveBox = responsive_box_1.DxResponsiveBox;
var sankey_1 = require("./sankey");
exports.DxSankey = sankey_1.DxSankey;
var scheduler_1 = require("./scheduler");
exports.DxScheduler = scheduler_1.DxScheduler;
var scroll_view_1 = require("./scroll-view");
exports.DxScrollView = scroll_view_1.DxScrollView;
var select_box_1 = require("./select-box");
exports.DxSelectBox = select_box_1.DxSelectBox;
var slide_out_1 = require("./slide-out");
exports.DxSlideOut = slide_out_1.DxSlideOut;
var slide_out_view_1 = require("./slide-out-view");
exports.DxSlideOutView = slide_out_view_1.DxSlideOutView;
var slider_1 = require("./slider");
exports.DxSlider = slider_1.DxSlider;
var sortable_1 = require("./sortable");
exports.DxSortable = sortable_1.DxSortable;
var sparkline_1 = require("./sparkline");
exports.DxSparkline = sparkline_1.DxSparkline;
var speed_dial_action_1 = require("./speed-dial-action");
exports.DxSpeedDialAction = speed_dial_action_1.DxSpeedDialAction;
var switch_1 = require("./switch");
exports.DxSwitch = switch_1.DxSwitch;
var tab_panel_1 = require("./tab-panel");
exports.DxTabPanel = tab_panel_1.DxTabPanel;
var tabs_1 = require("./tabs");
exports.DxTabs = tabs_1.DxTabs;
var tag_box_1 = require("./tag-box");
exports.DxTagBox = tag_box_1.DxTagBox;
var text_area_1 = require("./text-area");
exports.DxTextArea = text_area_1.DxTextArea;
var text_box_1 = require("./text-box");
exports.DxTextBox = text_box_1.DxTextBox;
var tile_view_1 = require("./tile-view");
exports.DxTileView = tile_view_1.DxTileView;
var toast_1 = require("./toast");
exports.DxToast = toast_1.DxToast;
var toolbar_1 = require("./toolbar");
exports.DxToolbar = toolbar_1.DxToolbar;
var tooltip_1 = require("./tooltip");
exports.DxTooltip = tooltip_1.DxTooltip;
var tree_list_1 = require("./tree-list");
exports.DxTreeList = tree_list_1.DxTreeList;
var tree_map_1 = require("./tree-map");
exports.DxTreeMap = tree_map_1.DxTreeMap;
var tree_view_1 = require("./tree-view");
exports.DxTreeView = tree_view_1.DxTreeView;
var validation_group_1 = require("./validation-group");
exports.DxValidationGroup = validation_group_1.DxValidationGroup;
var validation_summary_1 = require("./validation-summary");
exports.DxValidationSummary = validation_summary_1.DxValidationSummary;
var validator_1 = require("./validator");
exports.DxValidator = validator_1.DxValidator;
var vector_map_1 = require("./vector-map");
exports.DxVectorMap = vector_map_1.DxVectorMap;
