// devextreme-vue/speed-dial-action.js
/*!
 * devextreme-vue
 * Version: 20.1.6
 * Build date: Fri Jul 17 2020
 *
 * Copyright (c) 2012 - 2020 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var VueType = require("vue");
var Vue = VueType.default || VueType;
var speed_dial_action_1 = require("devextreme/ui/speed_dial_action");
var component_1 = require("./core/component");
var DxSpeedDialAction = Vue.extend({
    extends: component_1.DxComponent,
    props: {
        accessKey: String,
        activeStateEnabled: Boolean,
        elementAttr: Object,
        focusStateEnabled: Boolean,
        hint: String,
        hoverStateEnabled: Boolean,
        icon: String,
        index: Number,
        label: String,
        onClick: Function,
        onContentReady: Function,
        onDisposing: Function,
        onInitialized: Function,
        onOptionChanged: Function,
        rtlEnabled: Boolean,
        tabIndex: Number,
        visible: Boolean
    },
    computed: {
        instance: function () {
            return this.$_instance;
        }
    },
    beforeCreate: function () {
        this.$_WidgetClass = speed_dial_action_1.default;
    }
});
exports.DxSpeedDialAction = DxSpeedDialAction;
exports.default = DxSpeedDialAction;
