// devextreme-vue/core/configuration-component.js
/*!
 * devextreme-vue
 * Version: 20.1.6
 * Build date: Fri Jul 17 2020
 *
 * Copyright (c) 2012 - 2020 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var VueType = require("vue");
var Vue = VueType.default || VueType;
var configuration_1 = require("./configuration");
function getConfig(vueInstance) {
    if (!vueInstance.$vnode) {
        return;
    }
    var componentOptions = vueInstance.$vnode.componentOptions;
    return componentOptions && componentOptions.$_config;
}
exports.getConfig = getConfig;
function getInnerChanges(vueInstance) {
    if (!vueInstance.$vnode) {
        return;
    }
    var componentOptions = vueInstance.$vnode.componentOptions;
    return componentOptions && componentOptions.$_innerChanges;
}
exports.getInnerChanges = getInnerChanges;
function initOptionChangedFunc(config, vueInstance, innerChanges) {
    if (!config) {
        return;
    }
    config.init(Object.keys(vueInstance.$props));
    configuration_1.setEmitOptionChangedFunc(config, vueInstance, innerChanges);
}
exports.initOptionChangedFunc = initOptionChangedFunc;
function getComponentInfo(_a, removed) {
    var name = _a.name, isCollectionItem = _a.isCollectionItem, ownerConfig = _a.ownerConfig;
    var parentPath = ownerConfig && ownerConfig.fullPath;
    var optionPath = name && parentPath ? parentPath + "." + name : name || "";
    return {
        optionPath: optionPath,
        isCollection: isCollectionItem,
        removed: removed
    };
}
var DxConfiguration = Vue.extend({
    beforeMount: function () {
        var config = getConfig(this);
        var innerChanges = getInnerChanges(this);
        initOptionChangedFunc(config, this, innerChanges);
        configuration_1.bindOptionWatchers(config, this, innerChanges);
    },
    mounted: function () {
        if (this.$parent.$_instance) {
            this.$parent.$_config.componentsCountChanged
                .push(getComponentInfo(getConfig(this)));
        }
    },
    beforeDestroy: function () {
        this.$parent.$_config.componentsCountChanged
            .push(getComponentInfo(getConfig(this), true));
    },
    render: function (createElement) {
        return createElement();
    }
});
exports.DxConfiguration = DxConfiguration;
